### PR TITLE
Back end logic

### PR DESCRIPTION
**Pull request by vdequercize**
Usability update:
- Deactivated countries for which we have no data
- Capped the max journey length to 21 days as we don't have enough cities to populate longer journeys
- Deleted the bookmark object as we don't use it
- Added a little styling to the devise pages
- Added range plugin to flatpickr

### CODE REVIEW

File: app/assets/stylesheets/pages/_devise.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+    background-color:#4CAF50; /* Green */
+    border: none;
+    color: white !important;
+    padding: 5px 10px;
+    margin-bottom: 10px;
+    text-align: center;
+    text-decoration: none;
+    display: inline-block;
+    border-radius: 10px;
+  }
```
Comment (vdequercize) [inline app/assets/stylesheets/pages/_devise.scss:11]: Added a login-button class

File: app/assets/stylesheets/pages/_index.scss [hunk 2]
```diff
@@ -5,3 +5,4 @@
 @import "profile";
 @import "edit";
 @import "show";
+@import "devise";
```
Comment (vdequercize) [inline app/assets/stylesheets/pages/_index.scss:8]: Imported the previous scss page

File: app/javascript/controllers/flatpickr_controller.js [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 export default class extends Controller {
   static targets = [ "startDate", "endDate" ]
   connect() {
-    flatpickr(this.startDateTarget, {})
+    flatpickr(this.startDateTarget, {
+      plugins: [new rangePlugin({ input: "#end_date"})]
+    })
     flatpickr(this.endDateTarget, {})
   }
 }
```
Comment (vdequercize) [inline app/javascript/controllers/flatpickr_controller.js:13]: Added the range plugin

File: app/models/city.rb [hunk 4]
```diff
@@ -3,8 +3,6 @@ class City < ApplicationRecord
   geocoded_by :name
   after_validation :geocode, if: :will_save_change_to_name?
   # belongs_to :city_journey
-
-
   has_many :city_journeys, dependent: :destroy
   has_many :journeys, through: :city_journeys
   has_many :activities
```
Comment (vdequercize) [inline app/models/city.rb:8]: Deleted white spaces

File: app/models/journey.rb [hunk 5]
```diff
@@ -2,11 +2,16 @@ class Journey < ApplicationRecord
...
   validates :country, presence: true
   validates :tag, presence: true
+  validate :journey_length
 
+  def journey_length
+    if end_date - start_date > 22
+      errors.add(:end_date, "can't be more than 21 days after start date")
+    end
+  end
 end
```
Comment (vdequercize) [inline app/models/journey.rb:17]: Added max journey length validation rule

File: app/models/user.rb [hunk 6]
```diff
@@ -5,5 +5,4 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable
   has_many :journeys
   has_one_attached :photo
-  has_many :bookmarks
 end
```
Comment (vdequercize) [inline app/models/user.rb:8]: Deleted bookmark reference

File: app/views/devise/passwords/new.html.erb [hunk 7]
```diff
@@ -1,4 +1,4 @@
-<h2 class="mx-auto">Forgot your password?</h2>
+<h2 class="mx-auto mt-5">Forgot your password?</h2>
 
 <div class="d-flex justify-content-center border w-50 mx-auto">
   <%= simple_form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :post }) do |f| %>
```
Comment (vdequercize) [inline app/views/devise/passwords/new.html.erb:4]: Added margin at the top of the page

File: app/views/devise/registrations/new.html.erb [hunk 8]
```diff
@@ -23,22 +23,20 @@
...
-
     </div>
+  <% end %>
 </div>
 
-<% end %>
-
 <div class="mx-auto my-3">
   <%= link_to "Back", "/users/sign_in" %>
 </div>
```
Comment (vdequercize) [inline app/views/devise/registrations/new.html.erb:42]: Added margin, corrected syntax and added button class

File: app/views/devise/sessions/new.html.erb [hunk 9]
```diff
@@ -13,7 +13,7 @@
       <%= f.input :remember_me, as: :boolean if devise_mapping.rememberable? %>
     </div>
 
-    <div class="form-actions">
+    <div class="form-actions login-button">
       <%= f.button :submit, "Log in" %>
     </div>
   <% end %>
```
Comment (vdequercize) [inline app/views/devise/sessions/new.html.erb:19]: Added margin and button class

File: app/views/devise/unlocks/new.html.erb [hunk 10]
```diff
@@ -1,4 +1,4 @@
-<h2>Resend unlock instructions</h2>
+<h2 class="mt-5">Resend unlock instructions</h2>
 
 <%= simple_form_for(resource, as: resource_name, url: unlock_path(resource_name), html: { method: :post }) do |f| %>
   <%= f.error_notification %>
```
Comment (vdequercize) [inline app/views/devise/unlocks/new.html.erb:4]: Added margin

File: app/views/journeys/new.html.erb [hunk 11]
```diff
@@ -14,9 +14,9 @@
...
   <%= simple_form_for @journey, data: { controller: 'flatpickr' } do |f| %>
     <%= f.input :name, label: "Name Your Journey" %>
-    <%= f.input :country, label: "Where to?", priority: [ "France" ], collection: [ "Australia", "Brazil", "France", "New Zealand"] %>
+    <%= f.input :country, label: "Where to?", collection: [ "Australia (not available yet)", "Brazil (not available yet)", "France", "New Zealand (not available yet)"], disabled: ["Australia (not available yet)", "Brazil (not available yet)", "New Zealand (not available yet)"] %>
     <%= f.input :start_date, as: :string, input_html: { data: { flatpickr_target: "startDate" } } %>
-    <%= f.input :end_date, as: :string, input_html: { data: { flatpickr_target: "endDate" } } %>
+    <%= f.input :end_date, as: :string, input_html: { data: { flatpickr_target: "endDate" }, id: 'end_date' } %>
     <%= f.input :tag,
             as: :radio_buttons,
             collection_wrapper_tag: 'div',
```
Comment (vdequercize) [inline app/views/journeys/new.html.erb:22]: Deactivated countries for which we don't have data and added range plugin

File: config/routes.rb [hunk 12]
```diff
@@ -15,7 +15,6 @@
 
   resources :journeys
   resources :cities, only:[:index, :show]
-  resources :bookmarks, only:[:new, :create]
 
   get "/journeys/generate/new", to: "journeys#generate_new"
   post "/journeys/create/new", to: "journeys#create_new"
```
Comment (vdequercize) [inline config/routes.rb:20]: Deleted bookmark route

**Review by AJCJ1 — APPROVED** [review]
Comments from Victor have been noted.

**Pull request merged**